### PR TITLE
Add protractor tests for credentials.

### PR DESCRIPTION
**Pull request by lamara**

### CODE REVIEW

File: test/protractor/pages/credential.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  component.element(by.attribute('stackable-trigger', '$ctrl.menu')).click();
+  var actionMenu = element(by.tagName('br-credential-viewer-action-menu'));
+  actionMenu.element(by.partialLinkText('Edit')).click();
+  bedrock.waitForModalTransition();
+  var editModal = element(by.tagName('br-edit-credential-modal'));
+  var checkbox = editModal.element(by.model('model.allPublic'));
+  expect(checkbox.isSelected()).to.eventually.be.false;
+  checkbox.click();
+  editModal.element(by.partialButtonText('Save')).click();
+  browser.wait(function() {
```
Comment (mattcollier) [inline test/protractor/pages/credential.js]: browser.wait(EC.presenceOf(editModal), 8000);

Better?

File: test/protractor/pages/credential.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  component.element(by.attribute('stackable-trigger', '$ctrl.menu')).click();
+  var actionMenu = element(by.tagName('br-credential-viewer-action-menu'));
+  actionMenu.element(by.partialLinkText('Edit')).click();
+  bedrock.waitForModalTransition();
+  var editModal = element(by.tagName('br-edit-credential-modal'));
+  var checkbox = editModal.element(by.model('model.allPublic'));
+  expect(checkbox.isSelected()).to.eventually.be.true;
+  checkbox.click();
+  editModal.element(by.partialButtonText('Save')).click();
+  browser.wait(function() {
```
Comment (mattcollier) [inline test/protractor/pages/credential.js]: browser.wait(EC.presenceOf(editModal), 8000);

Better?

File: test/protractor/pages/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+var pages = GLOBAL.bedrock.pages || {};
```
Comment (mattcollier) [inline test/protractor/pages/index.js]: lowercase `global`

File: test/protractor/prepare.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+var bedrock = GLOBAL.bedrock;
```
Comment (mattcollier) [inline test/protractor/prepare.js]: lowercase global

File: test/protractor/pages/credential.js (hunk 2 above)
Comment (lamara) [inline test/protractor/pages/credential.js]: This is waiting for the modal to /not/ be present, which iirc isn't possible using the default expected conditions, so I had to use the custom promise below to return the negation of editModal.isPresent(). Definitely messy, and could definitely be better as a helper function (or possibly as a new expected condition -- is it possible to extend the EC class?)

Comment (mattcollier) [inline test/protractor/pages/credential.js]: Here's the docs: http://www.protractortest.org/#/api?view=ProtractorExpectedConditions

Don't know what kinds of behavior you encountered, but `editModal` _should_ just be an element like any other right?  I don't know why EC wouldn't work.

Comment (mattcollier) [inline test/protractor/pages/credential.js]: I will add that for a modal 'visibilityOf' might be a better choice.

Comment (lamara) [inline test/protractor/pages/credential.js]: EC.stalenessOf seems like what we want here. A weird name to call it and is probably why I missed it the first time.

File: test/protractor/pages/credential.js [hunk 5]
```diff
@@ -25,11 +25,7 @@ api.markPublic = function() {
   expect(checkbox.isSelected()).to.eventually.be.false;
   checkbox.click();
   editModal.element(by.partialButtonText('Save')).click();
-  browser.wait(function() {
-    return editModal.isPresent().then(function(res) {
-      return !res;
-    });
-  }, 8000);
+  browser.wait(EC.stalenessOf(editModal), 8000);
```
Comment (mattcollier) [inline test/protractor/pages/credential.js]: EC.invisibilityOf doesn't work here?

Comment (lamara) [inline test/protractor/pages/credential.js]: stalenessOf waits for the element to get detatched from the DOM, which is what we want here. invisibilityOf just waits for it to stop being displayed.

**Pull request merged**